packages/ember-metal/tests/chains_test.js

```diff
@@ -76,3 +76,14 @@ QUnit.test('observer and CP chains', function() {
   get(obj, 'qux'); // CP chain re-recreated
   ok(true, 'no crash');
 });
+
+QUnit.test('checks cache correctly', function(assert) {
+  let obj = { foo: {} };
+  let parentChainNode = new ChainNode(null, null, obj);
+  let chainNode = new ChainNode(parentChainNode, 'foo');
+
+  defineProperty(obj, 'foo', computed(function() { return UNDEFINED; }));
```

Comment:
This should return undefined, when it is `undefined` the cache will have `UNDEFINED` and the chain will descend when it should stop.